src/ERC4337Account.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IERC4337Account} from "./interfaces/IERC4337Account.sol";
+
+abstract contract ERC4337Account is IERC4337Account {
+    uint256 internal constant SIG_VALIDATION_FAILED = 1;
+    uint256 internal constant SIG_VALIDATION_SUCCEEDED = 0;
```

Comment:
nit: reverse the order of these?

Reply:
this is defined by the spec! "SHOULD not return early when returning SIG_VALIDATION_FAILED (1). Instead, it SHOULD complete the normal flow to enable performing a gas estimation for the validation function."  https://eips.ethereum.org/EIPS/eip-4337#smart-contract-account-interface

Reply:
ah yeah i just mean the order within the code, defining success before fail